src/libraries/System.Net.Security/src/System/Net/Security/TlsFrameHelper.cs

```diff
@@ -171,16 +230,55 @@ public static bool TryGetHandshakeInfo(ReadOnlySpan<byte> frame, ref TlsFrameHan
 
             info.SupportedVersions = info.Header.Version;
 
-            info.HandshakeType = (TlsHandshakeType)frame[5];
+            if (info.Header.Type == TlsContentType.Alert)
+            {
+                TlsAlertLevel level = default;
+                TlsAlertDescription description = default;
+                if (TryGetAlertInfo(frame, ref level, ref description))
+                {
+                    info.AlertDescription = description;
```

Comment:
Why don't you save the alert level info?

Reply:
I was not sure if that would be ever useful. It is only byte so we could for completeness...

Reply:
"In TLS 1.3, the severity is implicit in the type of alert being sent, and the "level" field can safely be ignored" from (RFC 8446 Section 6)[https://tools.ietf.org/html/rfc8446#section-6]